cli/cmd/proxy/read/command_test.go

```diff
@@ -146,6 +140,41 @@ func TestReadCommandOutput(t *testing.T) {
 	}
 }
 
+// TestFilterWarnings ensures that a warning is printed if the user applies a
+// field filter (e.g. -fqdn default) and a table filter (e.g. -secrets) where
+// the former does not affect the output of the latter.
+func TestFilterWarnings(t *testing.T) {
+	podName := "fakePod"
+	cases := map[string][]string{
+		"fully qualified domain name doesn't apply to secrets":                {"-fqdn", "default", "-secrets"},
+		"port doesn't apply to secrets or routes":                             {"-port", "8080", "-secrets", "-routes"},
+		"fully qualified domain name doesn't apply to endpoints or listeners": {"-fqdn", "default", "-endpoints", "-listeners"},
```

Comment:
I think we need test cases for:
- `-address` flag warning
- all possible combindations when `-fqdn` flag would produce a warning
- all possible combinations when `-port` flag would produce a warning
- a case when mutliple warnings are returned
- happy case scenarios when no warnings are returned (if that's not tested elsewhere).